src/Symfony/Component/Security/Core/User/UserInterface.php

```diff
@@ -26,6 +26,8 @@
  *
  * @see UserProviderInterface
  *
+ * @method string getUserIdentifier() returns the identifier for this user (e.g. its username or e-mailaddress)
```

Comment:
`getIdentifier`?

i think only token needs to convey it's the user identifier

Reply:
90% of the time (honestly, I think more), people are using entities as users. I don't want them to confuse `getIdentifier()` as `getId()`: your users don't login with a uuid, but with some other identifier (e-mailaddres). See also https://github.com/symfony/symfony/pull/40403#pullrequestreview-610508853

Reply:
in that regard, i think getId vs getUserIdentifier is still as confusing

Reply:
also AFAIK doctrine entities actually forwarding getIdentifier to getId would still be legit. (even while logging in by email)